### ISSUE TITLE
Version catalogues: allow nested bundles

### ISSUE THREAD

**Issue by jshiell** [gradle/gradle#17501]
Hi folks. I haven't found anything in the [docs](https://docs.gradle.org/current/userguide/platforms.html) and the [FAQ](https://melix.github.io/blog/2021/03/version-catalogs-faq.html#_why_can_t_i_use_excludes_or_classifiers) discussing composition of bundles in the version catalogue, so I'd like to suggest this as a feature request.

Our use case is that we have bundles of depedencies that are supersets of indepdentently useful bundles. For example, we may wish to apply a `jettyApi` bundle to library module, but a `jettyServer` bundle - which is a superset of `jettyApi` - to web application modules.

As a concrete example, we have a TOML with a `junitJupiterVintage` group that references the `junitJupiter` group. We'd like our modules to be pure Jupiter, but while they're migrated it's useful to have a superset with the vintage libraries, and avoid having to repeat ourselves in either the TOML or the individual module build files.

```
[versions]
junit_platform = "1.7.2"
junit_jupiter = "5.7.2"

[libraries]
apiguardian-api = { group = "org.apiguardian", name = "apiguardian-api", version = "1.1.0" }
opentest4j = { group = "org.opentest4j", name = "opentest4j", version = "1.2.0" }
junit-platform-commons = { group = "org.junit.platform", name = "junit-platform-commons", version.ref = "junit_platform" }
junit-platform-launcher = { group = "org.junit.platform", name = "junit-platform-launcher", version.ref = "junit_platform" }
junit-jupiter-api = { group = "org.junit.jupiter", name = "junit-jupiter-api", version.ref = "junit_jupiter" }
junit-jupiter-params = { group = "org.junit.jupiter", name = "junit-jupiter-params", version.ref = "junit_jupiter" }

junit = { group = "junit", name = "junit", version = "4.13.2" }

[bundles]

junitJupiter = [
    "apiguardian-api",
    "opentest4j",
    "junit-platform-commons",
    "junit-platform-launcher",
    "junit-jupiter-api",
    "junit-jupiter-params",
]

junitJupiterVintage = [
    "junitJupiter",
    "junit"
]
```

At present in 7.1 such a construct results in a pretty descriptive error:

```
org.gradle.api.InvalidUserDataException: Invalid catalog definition:
  - Problem: In version catalog libs, a bundle with name 'junitJupiterVintage' declares a dependency on 'junitJupiter' which doesn't exist.

    Reason: Bundles can only contain references to existing library aliases.

    Possible solutions:
      1. Make sure that the library alias 'junitJupiter' is declared.
      2. Remove 'junitJupiter' from bundle 'junitJupiterVintage'.
...
```

Thanks muchly!

**Comment by melix** [comment]
If I understand properly what you're looking after is not nested bundles (bundles which are accessed via nested accessors, e.g `libs.bundles.top.middle.bottom`, but _composition of bundles_ (bundles which include the contents of another bundle). Correct?

**Comment by jshiell** [comment]
Yes, exactly - "bundles which include the contents of another bundle" is a very concise summary.

**Comment by mvignale1987** [comment]
This!
I believe that compositing bundle contents could be super beneficial for concise and effective dependency declarations.

There is a workaround for this that involves replicating the bundle's content on every subsequent bundle, but that grows with the bundle's size and is prone to errors, so really looking after this if possible.

Thanks!

**Comment by LeoColman** [comment]
To demonstrate this ~~and sneakily bump the issue~~  I have another real case that would benefit from this: Android Testing


Kotest has a runner for Android and a runner for the JVM.
All other dependencies are the same.

The DRY alternative to it:
```
[bundles]
kotest-android = ["kotest-runner-android", "kotest-assertions-android"]
kotest-jvm = ["kotest-runner-junit5"]
kotest-common = ["kotest-property", "kotest-assertions"]

```

The Do-RY alternative to it:
```
[bundles]
kotest-android = ["kotest-runner-android", "kotest-assertions-android", "kotest-property", "kotest-assertions"]
kotest-jvm = ["kotest-runner-junit5", "kotest-property", "kotest-assertions"]
```


The ideal (and point of this issue) option:
```
[bundles]
kotest-android = ["kotest-runner-android", "kotest-assertions-android", "kotest-common"]
kotest-jvm = ["kotest-runner-junit5", "kotest-common"]
kotest-common = ["kotest-property", "kotest-assertions"]
```


It's similar to what we can already achieve with the DRY alternative. However it shortens what needs to be added to the `build.gradle.kts`:

```
testImplementation(libs.bundles.kotest.jvm)
testImplementation(libs.bundles.kotest.common)

androidTestImplementation(libs.bundles.kotest.android)
androidTestImplementation(libs.bundles.kotest.common)
```

to

```
testImplementation(libs.bundles.kotest.jvm)

androidTestImplementation(libs.bundles.kotest.android)
```

IMO this is better mainly to reduce the amount of code added to `build.gradle.kts`

**Comment by Romanow88** [comment]
Would be really awesome if this gets implemented. Seems like a low hanging fruit with a pretty bug impact.

For us this is currently a pain point with moving to version catalogues for a project with about 300 modules / subprojects.

**Comment by meredrica** [comment]
this would indeed be nice to have since frameworks like micronaut come with a lot of aliases already and we would like to group them.

**Comment by melix** [comment]
> this would indeed be nice to have since frameworks like micronaut come with a lot of aliases already and we would like to group them.

The next version of the Micronaut Gradle plugin will let you automatically import the Micronaut catalog as `mn`, so it's already namespaced. Even now, I'd recommend it to import it into its own namespace instead of `libs`.

**Comment by MnsrIl** [comment]
Have to upvote this one. Will we see some solution soon? I see these as really useful feature, which would make a lot easier defining dependency bundles when you have several submodules, which require usually same bundles everywhere less or more
